### PR TITLE
convert to jenkins ci

### PR DESCRIPTION
**Pull request by maratoid**

### CODE REVIEW

File: charts/technical-on-boarding/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          valueFrom:
+            secretKeyRef:
+              name: {{ template "fullname" . }}
+              key: ONBOARD_CLIENT_SECRET
+        - name: ONBOARD_ORG
+          valueFrom:
+            configMapKeyRef:
+              name: {{ template "fullname" . }}
+              key: ONBOARD_ORG
+        - name: ONBOARD_REPO
```
Comment (samba) [inline charts/technical-on-boarding/templates/deployment.yaml]: We'll need to add an entry here for `ONBOARD_CLIENT_ENDPOINT` as per #25.

File: pipeline.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+type: chart
+rootfs:
+  - image: samsung_cnct/technical-on-boarding-container
+    buildArgs: 
+      - arg: VERSION
+        value: 1.1.7
```
Comment (samba) [inline pipeline.yaml]: Should this be hard-coded here? Does it get overridden by the `.versionfile`?

File: pipeline.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    test: 
+      values:
+        - key: ingress
+          value: "false"
+    stage: 
+      values:
+        - key: ingress
+          value: "false"  
+prod: 
+  doDeploy: none
```
Comment (samba) [inline pipeline.yaml]: This project is still alive, still improving and still deployed in prod. Is this `none` here suitable? 

Previously we had it deploying to `staging` on all merges into `master`, and into `prod` on all git version-tags. ([as seen here][1])
How should we facilitate a suitable workflow for this application?


[1]: https://github.com/samsung-cnct/chart-technical-on-boarding/blob/master/.gitlab-ci.yml

**Comment by guineveresaenger** [comment]
@samba @maratoid noted. I'm looking forward to getting all this integrated.

File: pipeline.yaml (hunk 2 above)
Comment (maratoid) [reply pipeline.yaml]: no other way to do this with build args right now. i would suggest maybe moving 
```# set version
RUN sed -i -- 's/${VERSION}/'"$VERSION"'/g' conf/app.conf && \
	  sed -i -- 's/${BUILD}/'"$BUILD"'/g' conf/app.conf```

to a config map template in chart? and then using `{{.Chart.Version}}`

File: pipeline.yaml (hunk 3 above)
Comment (maratoid) [reply pipeline.yaml]: Change the pipeline.yaml in this way:

```
type: chart
rootfs:
  - image: samsung_cnct/technical-on-boarding-container
    buildArgs: 
      - arg: VERSION
        value: 1.1.7
      - arg: BUILD
        env: GIT_COMMIT
    context: technical-on-boarding-container
    dockerContext: .
    chart: technical-on-boarding
    value: image
configs:
  - chart: technical-on-boarding
    timeout: 600
    retries: 1
    release: onboarding
    test: 
      values:
        - key: ingress
          value: "false"
    stage:
      values:
        - staging values here
    prod:
      values:
        - prod values here
    stage: 
      values:
        - key: ingress
          value: "false"
stage: 
  deploy: true
prod: 
  doDeploy: versionfile```

per https://github.com/samsung-cnct/pipeline-library#cnct-shared-workflow-library

**Comment by samba** [comment]
Note for self; CI in GitLab is failing because the Dockerfile moved, among other things.

File: pipeline.yaml (hunk 2 above)
Comment (samba) [reply pipeline.yaml]: This is resolved in #31 by way of carefully placed environment variables, based on the chart templates and existing CI build-time features. 

The configmap approach made some sense, but would have made local dev/testing more complex.

File: charts/technical-on-boarding/templates/deployment.yaml (hunk 1 above)
Comment (samba) [reply charts/technical-on-boarding/templates/deployment.yaml]: Resolving in #31 (commit 309d9ee), with accompanying values file defaults, etc.

File: pipeline.yaml (hunk 3 above)
Comment (samba) [reply pipeline.yaml]: These considerations will be applied in #31. (WIP)

**Comment by samba** [comment]
After discussion with @maratoid, closing in favor of #31. 
Thanks for your help in getting this started, @maratoid !

**Pull request closed without merging**